packages/studio-ui/src/components/AddPageButton/AddPageContext.ts

```diff
@@ -1,14 +1,18 @@
-import { StreamScope } from "@yext/studio-plugin";
+import { GetPathVal, StreamScope } from "@yext/studio-plugin";
 import { createContext } from "react";
 
 export interface AddPageData {
```

Comment:
I'm also not wild about how we do `AddPageData`. It doesn't make a lot of sense to me that `AddPageData` doesn't contain all the info necessary to add a page. Before we added Layouts, it was missing name and getPathVal. Now it has those but is missing Layouts. I think this is something we should rework too, but not in this PR! The pattern was introduced before this PR.

Reply:
For sure, I agree its not super great. I tried to break out of this pattern, but the reason I had to keep along with it was because even if we add layoutState to `AddPageData` and set it once the user decides on a layout, the context does not update in time for it to be used by `createPage`.